From e23818e09ab1fa31534761d5b008c48c33ff88d3 Mon Sep 17 00:00:00 2001
From: Joseph Myers <joseph@codesourcery.com>
Date: Tue, 9 Aug 2016 14:25:25 -0700
Subject: [PATCH 03/16] Merge AF_BUS changes.

Value same as previously used in CodeBench, but will need changing in
future (Linux now uses that value for AF_QIPCRTR, as of Linux 4.7).

	Merge from SCB glibc 2.23:

        2013-06-17  Meador Inge  <meadori@codesourcery.com>

        CB-2411

        * Makefile (installed_headers): Add socket/sys/bus.h.
        * sysdeps/unix/sysv/linux/bits/socket.h (PF_BUS, AF_BUS): New
        macros.
        * include/sys/bus.h: New file.
        * socket/sys/bus.h: Likewise.
        * socket/sys/socket.h (__SOCKADDR_ALLTYPES): Add sockaddr_bus.
        * socket/Makefile (headers): Add sys/bus.h.
        * release-notes-csl.xml: Add a release note.

        2013-11-14  Meador Inge  <meadori@codesourcery.com>

        CB-2470

        * sysdeps/unix/sysv/linux/bits/socket.h (PF_BUS): Bump to 41.
        (PF_MAX): Bump to 42.

        2014-03-25  Meador Inge  <meadori@codesourcery.com>

        CB-3023

        * sysdeps/unix/sysv/linux/bits/socket.h (PF_BUS): Bump to 42.
        (PF_MAX): Bump to 43.

        2014-04-09  Meador Inge  <meadori@codesourcery.com>

        CB-3023

        * socket/sys/bus.h (SOL_BUS): Bump to 282.
---
 ChangeLog.csl                         | 37 +++++++++++++++++++++++
 Makefile                              |  2 +-
 include/sys/bus.h                     |  1 +
 socket/Makefile                       |  2 +-
 socket/sys/bus.h                      | 57 +++++++++++++++++++++++++++++++++++
 socket/sys/socket.h                   |  3 +-
 sysdeps/unix/sysv/linux/bits/socket.h |  4 ++-
 7 files changed, 102 insertions(+), 4 deletions(-)
 create mode 100644 include/sys/bus.h
 create mode 100644 socket/sys/bus.h

diff --git a/ChangeLog.csl b/ChangeLog.csl
index cc98729..4b42e70 100644
--- a/ChangeLog.csl
+++ b/ChangeLog.csl
@@ -2,6 +2,43 @@
 
 	Merge from SCB glibc 2.23:
 
+        2013-06-17  Meador Inge  <meadori@codesourcery.com>
+
+        CB-2411
+
+        * Makefile (installed_headers): Add socket/sys/bus.h.
+        * sysdeps/unix/sysv/linux/bits/socket.h (PF_BUS, AF_BUS): New
+        macros.
+        * include/sys/bus.h: New file.
+        * socket/sys/bus.h: Likewise.
+        * socket/sys/socket.h (__SOCKADDR_ALLTYPES): Add sockaddr_bus.
+        * socket/Makefile (headers): Add sys/bus.h.
+        * release-notes-csl.xml: Add a release note.
+
+        2013-11-14  Meador Inge  <meadori@codesourcery.com>
+
+        CB-2470
+
+        * sysdeps/unix/sysv/linux/bits/socket.h (PF_BUS): Bump to 41.
+        (PF_MAX): Bump to 42.
+
+        2014-03-25  Meador Inge  <meadori@codesourcery.com>
+
+        CB-3023
+
+        * sysdeps/unix/sysv/linux/bits/socket.h (PF_BUS): Bump to 42.
+        (PF_MAX): Bump to 43.
+
+        2014-04-09  Meador Inge  <meadori@codesourcery.com>
+
+        CB-3023
+
+        * socket/sys/bus.h (SOL_BUS): Bump to 282.
+
+2016-08-09  Joseph Myers  <joseph@codesourcery.com>
+
+	Merge from SCB glibc 2.23:
+
 	2016-02-27  Sandra Loosemore  <sandra@codesourcery.com>
 
 	CB-1356
diff --git a/Makefile b/Makefile
index 32748b3..6d6468f 100644
--- a/Makefile
+++ b/Makefile
@@ -297,7 +297,7 @@ installed-headers = argp/argp.h assert/assert.h catgets/nl_types.h \
 		    resource/sys/vtimes.h resource/ulimit.h rt/aio.h \
 		    rt/mqueue.h setjmp/setjmp.h shadow/shadow.h \
 		    signal/signal.h signal/sys/signal.h socket/sys/socket.h \
-		    socket/sys/un.h stdio-common/printf.h \
+		    socket/sys/un.h socket/sys/bus.h stdio-common/printf.h \
 		    stdio-common/stdio_ext.h stdlib/stdlib.h stdlib/alloca.h \
 		    stdlib/monetary.h stdlib/fmtmsg.h stdlib/ucontext.h \
 		    sysdeps/generic/inttypes.h sysdeps/generic/stdint.h \
diff --git a/include/sys/bus.h b/include/sys/bus.h
new file mode 100644
index 0000000..b635566
--- /dev/null
+++ b/include/sys/bus.h
@@ -0,0 +1 @@
+#include <socket/sys/bus.h>
diff --git a/socket/Makefile b/socket/Makefile
index 92450e8..ac7f369 100644
--- a/socket/Makefile
+++ b/socket/Makefile
@@ -22,7 +22,7 @@ subdir	:= socket
 
 include ../Makeconfig
 
-headers	:= sys/socket.h sys/un.h bits/sockaddr.h bits/socket.h \
+headers	:= sys/socket.h sys/bus.h sys/un.h bits/sockaddr.h bits/socket.h \
 	   bits/socket2.h sys/socketvar.h net/if.h
 
 routines := accept bind connect getpeername getsockname getsockopt	\
diff --git a/socket/sys/bus.h b/socket/sys/bus.h
new file mode 100644
index 0000000..f3a0082
--- /dev/null
+++ b/socket/sys/bus.h
@@ -0,0 +1,57 @@
+/* Copyright (C) 1991, 1995, 1996, 2001 Free Software Foundation, Inc.
+   This file is part of the GNU C Library.
+
+   The GNU C Library is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Lesser General Public
+   License as published by the Free Software Foundation; either
+   version 2.1 of the License, or (at your option) any later version.
+
+   The GNU C Library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Lesser General Public License for more details.
+
+   You should have received a copy of the GNU Lesser General Public
+   License along with the GNU C Library; if not, see
+   <http://www.gnu.org/licenses/>.  */
+
+#ifndef _SYS_BUS_H
+#define _SYS_BUS_H      1
+
+#include <sys/cdefs.h>
+
+/* Get the definition of the macro to define the common sockaddr members.  */
+#include <bits/sockaddr.h>
+
+__BEGIN_DECLS
+
+/* 'protocol' to use in socket(AF_BUS, SOCK_SEQPACKET, protocol) */
+#define BUS_PROTO_NONE  0
+#define BUS_PROTO_DBUS  1
+#define BUS_PROTO_MAX   1
+
+/* setsockopt() operations */
+#define SOL_BUS         282
+#define BUS_ADD_ADDR        1
+#define BUS_JOIN_BUS        2
+#define BUS_DEL_ADDR        3
+#define BUS_SET_EAVESDROP   4
+#define BUS_UNSET_EAVESDROP 5
+
+/* Bus address */
+struct bus_addr
+  {
+    __uint64_t s_addr; /* 16-bit prefix + 48-bit client address */
+  };
+
+/* Structure describing an AF_BUS socket address. */
+struct sockaddr_bus
+  {
+    __SOCKADDR_COMMON (sbus_);    /* AF_BUS */
+    struct bus_addr    sbus_addr; /* bus address */
+    char sbus_path[108];          /* pathname */
+  };
+
+__END_DECLS
+
+#endif /* sys/bus.h */
diff --git a/socket/sys/socket.h b/socket/sys/socket.h
index c9f0f50..0d3aa2a 100644
--- a/socket/sys/socket.h
+++ b/socket/sys/socket.h
@@ -84,7 +84,8 @@ enum
   __SOCKADDR_ONETYPE (sockaddr_iso) \
   __SOCKADDR_ONETYPE (sockaddr_ns) \
   __SOCKADDR_ONETYPE (sockaddr_un) \
-  __SOCKADDR_ONETYPE (sockaddr_x25)
+  __SOCKADDR_ONETYPE (sockaddr_x25) \
+  __SOCKADDR_ONETYPE (sockaddr_bus)
 
 # define __SOCKADDR_ONETYPE(type) struct type *__restrict __##type##__;
 typedef union { __SOCKADDR_ALLTYPES
diff --git a/sysdeps/unix/sysv/linux/bits/socket.h b/sysdeps/unix/sysv/linux/bits/socket.h
index 2eb95f7..fe3d05d 100644
--- a/sysdeps/unix/sysv/linux/bits/socket.h
+++ b/sysdeps/unix/sysv/linux/bits/socket.h
@@ -83,7 +83,8 @@ typedef __socklen_t socklen_t;
 #define PF_NFC		39	/* NFC sockets.  */
 #define PF_VSOCK	40	/* vSockets.  */
 #define PF_KCM		41	/* Kernel Connection Multiplexor.  */
-#define PF_MAX		42	/* For now..  */
+#define PF_BUS		42	/* AF_BUS sockets */
+#define PF_MAX		43	/* For now..  */
 
 /* Address families.  */
 #define AF_UNSPEC	PF_UNSPEC
@@ -131,6 +132,7 @@ typedef __socklen_t socklen_t;
 #define AF_NFC		PF_NFC
 #define AF_VSOCK	PF_VSOCK
 #define AF_KCM		PF_KCM
+#define AF_BUS		PF_BUS
 #define AF_MAX		PF_MAX
 
 /* Socket level values.  Others are defined in the appropriate headers.
-- 
2.8.1

